### PR TITLE
Fix #162: Add support for Digest Auth to RestClient

### LINKED ISSUE #162: Add support for Digest Auth to RestClient
See https://github.com/spring-projects/spring-framework/issues/24425

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java [hunk 1]
```diff
@@ -193,6 +200,12 @@ private void initializeWebClient() throws RestClientException {
         webClient = builder.baseUrl(config.getBaseUrl()).clientConnector(connector).build();
     }
 
+    private static void validateConfiguration(final RestClientConfiguration config) throws RestClientException {
+        if (config.isHttpBasicAuthEnabled() && config.isHttpDigestAuthEnabled()) {
```
Comment (banterCZ) [inline rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java:204]: Just a hand-brake to have reasonable configuration and not try to configure both.

File: rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java [hunk 2]
```diff
@@ -737,6 +746,33 @@ void testRedirectShouldFollowWhenEnabled() throws Exception {
...
+    void testGetWithResponseDigest() throws RestClientException {
+        final RestClientConfiguration config = prepareConfiguration();
+        config.setBaseUrl(privateBaseUrl);
+        config.setHttpBasicAuthEnabled(false);
+        config.setHttpDigestAuthEnabled(true);
+        config.setHttpDigestAuthUsername("test-digest-user");
+        config.setHttpDigestAuthPassword("top-secret");
+        final RestClient restClient = new DefaultRestClient(config);
+
+        ResponseEntity<Response> responseEntity = restClient.get("/response", new ParameterizedTypeReference<Response>() {});
```
Comment (banterCZ) [inline rest-client-base/src/test/java/com/wultra/core/rest/client/base/DefaultRestClientTest.java:759]: Double-checked the behavior. In the log, you may see:
```
WARN 51990 --- [ctor-http-nio-2] c.w.c.r.client.base.DefaultRestClient    : RestClient GET https://localhost:63921/private/api/test/response: 401 UNAUTHORIZED
INFO 51990 --- [ctor-http-nio-2] c.w.c.r.client.base.DefaultRestClient    : RestClient GET https://localhost:63921/private/api/test/response: 200 OK
```

File: rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java (hunk 1 above)
Comment (romanstrobl) [reply rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java:204]: Good idea, I just want to add comment about this, but it is already done. 👍

**Review by romanstrobl — APPROVED** [review]
Nice job! Let's watch the `netty-http-authenticator` library to make sure it gets regular updates.

**Pull request merged**